Remove pagination, sort dict entries

novellac opened the pull request:
## This PR fixes...
This PR fixes #79 . The dictionary wasn't sorting in a case-insensitive way.

## What I did...
- Removed the sorting from the graphql query (we no longer need it because it seems it can't be case-insensitive)
- Removed pagination. Pagination depended on the graphql query, so it we're going to use pagination it will have to be a bit different.
- Removed `pageInfo` from the graphql query. We're not paginating so it wouldn't return anything.
- Added a `data` property for holding sorted entries
- Created a method to sort the array of data and used it in the `created` hook

## How to test...

Navigate to /dictionary. Check that:
1. All dictionary entries are now printed on one page
1. Entries are sorted in alphabetical order _whether or not they start with upper/lowercase letter_.

## I learned...

GraphQL has some limitations, and also I need to learn more about it! 😆 😭

Commit:
Remove pagination, sort dict entries

ghost commented:
- Some words have the verbiage of "Replace with your own definition" (ex: Avatar, Backlog, Brand Book, etc.), but when I click on the word it's not sending me to Github to add anything. Instead, I only have the option of clicking "Back to Dictionary" and the page appears to load half way (see screenshot link below). 

Should there be a definition listed already listed on these or a screen to edit the pre existing definition?

<img width="1680" alt="Screen Shot 2020-10-07 at 7 43 45 PM" src="https://user-images.githubusercontent.com/70661426/95399113-6de96280-08d5-11eb-9a80-2ec031f5ab06.png">

- When I click on "Back to Dictionary" this brings me back to the beginning of the dictionary. I'm not sure if the logic was built that way on purpose, but while testing I have to go through all the pages or remember which word I left off on. 

- Pages 6 and 7 list End User and End Users, seems to be a duplicate entry for the same thing.

- Pages 8 and 9 list Heat Map and Heat Maps, seems to be a duplicate entry for the same thing.

- Pages 17-27 start with lowercase words in alphabetical order. Should these be in alphabetical order with the capital letters first then lowercase after (see example below) or as is with capital letters first in alphabetical order and then the lowercase letters in alphabetical order at the end?

Ex:
1. Abstract
2. agile
3. analytics
4. App

- I'm a bit confused on the printing on one page portion. Should all the words print just on one page in print preview? If so, this is how the print preview looks for me:

<img width="1125" alt="Screen Shot 2020-10-07 at 9 24 22 PM" src="https://user-images.githubusercontent.com/70661426/95404589-79dc2100-08e3-11eb-9fac-1cc8c4ffa5bc.png">

novellac commented:
> * Some words have the verbiage of "Replace with your own definition" (ex: Avatar, Backlog, Brand Book, etc.), but when I click on the word it's not sending me to Github to add anything. Instead, I only have the option of clicking "Back to Dictionary" and the page appears to load half way (see screenshot link below).
The page loads all the way, but the background color doesn't extend to the bottom of the screen (which isn't a good thing - it needs to be done still.).
> Should there be a definition listed already listed on these or a screen to edit the pre existing definition?

Yes there should be a link to edit the definition (it doesn't exist yet). This is because of a failed administrative experiment on my part 😬 . In Forestry, we were trying to get some better messaging for the contributors by having "Replace with your own definition" be the default text. It really backfired, because now all the words have definitions, but not the ones we want! Long story short, the _code_ is working properly, but our admin process has a bug to fix! We'll need to open another card for that.


> * When I click on "Back to Dictionary" this brings me back to the beginning of the dictionary. I'm not sure if the logic was built that way on purpose, but while testing I have to go through all the pages or remember which word I left off on.

This shouldn't be an issue anymore, since we removed the pagination. It will still go to the top of the page, but I think that's ok (though if it's a feature you think would be valuable, let's discuss and make a card for it!).

> * Pages 6 and 7 list End User and End Users, seems to be a duplicate entry for the same thing.
> * Pages 8 and 9 list Heat Map and Heat Maps, seems to be a duplicate entry for the same thing.

It looks like people have created a couple of extra definitions. We can delete these, but they should in their own card, since it's a bit outside the scope of this work.

> * Pages 17-27 start with lowercase words in alphabetical order. Should these be in alphabetical order with the capital letters first then lowercase after (see example below) or as is with capital letters first in alphabetical order and then the lowercase letters in alphabetical order at the end?

This shouldn't be an issue, since we've removed the pagination, they should all be printed together.

> * I'm a bit confused on the printing on one page portion. Should all the words print just on one page in print preview? If so, this is how the print preview looks for me:

We're not actually aiming at printing anything to paper. I think I see (let me know if I'm wrong, though!) where it could be confusing, though - when I use the word "print," I mean it should show up on the website, but not that you should have to print it out. I should have explained that earlier, sorry!

After re-reading all of these comments again, I think you may have been looking at the live site? The testing needs to be done on the deploy preview environment. To get to the preview envrionment, click on the "details" link on the very bottom row of the checks area (screenshot below).
![Remove-pagination-sort-dict-entries-by-novellac-·-Pull-Request-80-·-cherryontech-website](https://user-images.githubusercontent.com/38117965/95519333-e2360b80-0992-11eb-8185-0b20179bb108.png)

NiaCLay reviewed (commented):
I left some questions as comments.

ghost commented:
Re-reviewed using the correct deploy preview below. This looks all good now and I added 2 separate issues for the duplicate entries for "End User" and "Heat Map"

https://deploy-preview-80--cherryontech.netlify.app/dictionary/

NiaCLay reviewed (approved):
Re reviewed with correct deploy link. See comments.
